prow/cmd/branchprotector/protect.go

```diff
@@ -63,28 +63,8 @@ func gatherOptions() options {
 	return o
 }
 
-func jobRequirements(jobs []config.Presubmit, after bool) []string {
-	var required []string
-	for _, j := range jobs {
-		// Does this job require a context or have kids that might need one?
-		if !after && !j.AlwaysRun && j.RunIfChanged == "" {
-			continue // No
-		}
-		if !j.SkipReport { // This job needs a context
-			required = append(required, j.Context)
-		}
-		// Check which children require contexts
-		required = append(required, jobRequirements(j.RunAfterSuccess, true)...)
-	}
-	return required
-}
-
-func repoRequirements(org, repo string, cfg config.Config) []string {
-	p, ok := cfg.Presubmits[org+"/"+repo]
-	if !ok {
-		return nil
-	}
-	return jobRequirements(p, false)
+type ref struct {
+	org, repo, branch string
```

Comment:
Is golang smart enough to consider two types with equal fields equal?

This would require more work in python

Reply:
are you refering to tmpRefs := map[orgRepo][]string{} ?